lib/fastlane/plugin/wpmreleasetoolkit/helper/android/android_version_helper.rb

```diff
@@ -127,81 +265,162 @@ def self.bump_version_release
-                  return line unless line.include?("\"#{keyword}\"") or line.include?("P#{keyword}")
+                if line.include?(keyword) && !line.include?("\"#{keyword}\"") && !line.include?("P#{keyword}")
+                  return line.split(' ')[1]
                 end
               end
             end
           end
           return nil
         end
 
+        # Ensure that a version string is correctly formatted (that is, each of its parts is a number) and returns the 2-parts version number
+        #
+        # @param [String] version The version string to verify
+        #
+        # @return [String] The "major.minor" version string, only with the first 2 components
+        # @raise [UserError] If any of the parts of the version string is not a number
+        #
         def self.verify_version(version)
           v_parts = get_version_parts(version)
           
           v_parts.each do | part |
-            if (!is_number?(part)) then
+            if (!is_int?(part)) then
               UI.user_error!("Version value can only contains numbers.")
             end
           end
 
           "#{v_parts[MAJOR_NUMBER]}.#{v_parts[MINOR_NUMBER]}"
         end
 
-        def self.is_number? string
-          true if Float(string) rescue false
+        # Check if a string is an integer.
+        #
+        # @param [String] string The string to test
+        #
+        # @return [Bool] true if the string is representing an integer value, false if not
+        #
+        def self.is_int? string
+          true if Integer(string) rescue false
         end
 
+        # The path to the build.gradle file for the project.
+        #
+        # @env PROJECT_ROOT_FOLDER The path to the root of the project (the folder containing the `.git` directory).
+        # @env PROJECT_NAME The name of the project, i.e. the name of the subdirectory containing the project's `build.gradle` file.
+        #
+        # @return [String] The path of the `build.gradle` file inside the project subfolder in the project's repo
+        #
         def self.gradle_path 
-          "#{ENV["PROJECT_ROOT_FOLDER"]}#{ENV["PROJECT_NAME"]}/build.gradle"
+          File.join(ENV["PROJECT_ROOT_FOLDER"] || '.', ENV["PROJECT_NAME"], 'build.gradle')
         end
 
-        # FIXME: This implementation is very fragile. This should be done parsing the file in a proper way. 
-        # Leveraging gradle itself is probably the easiest way.
+        # Update both the versionName and versionCode of the build.gradle file to the specified version.
+        #
+        # @param [Hash] version The version hash, containing values for keys "name" and "code"
+        # @param [String] section The name of the section to update in the build.gradle file, e.g. "defaultConfig" or "vanilla"
+        #
+        # @todo This implementation is very fragile. This should be done parsing the file in a proper way. 
+        #       Leveraging gradle itself is probably the easiest way.
```

Comment:
> Leveraging gradle itself is probably the easiest way.

👍

Reply:
FWIW I'm not even sure I'll go down that rabbit hole during my cleanup project though. This FIXME has been here for a while and I've looked quickly at how to do it on the net and all solutions involve modifying the .gradle files themselves to add a task that would print the value. Which would make our lanes depend on the presence of such task, and would not solve the other similar method and technique we use when writing/updating the version name and code… so probably gonna keep it there for a while still.